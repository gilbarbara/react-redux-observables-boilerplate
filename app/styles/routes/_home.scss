.app__home {
  padding: 0;

  &__intro .app__container {
    align-items: center;
    background-color: $app-color-alt;
    color: #fff;
    display: flex;
    flex-direction: column;
    justify-content: center;
    min-height: calc(100vh - #{$header-height});

    h1 {
      color: #fff;
      font-size: $font-size-h3;
      line-height: 1.4;
      margin: 0 auto ($grid-gutter-width / 3);
      max-width: 52rem;
      text-align: center;

      @include sm {
        font-size: $font-size-h2;
      }
    }

    p {
      margin-bottom: $grid-gutter-width;
    }
  }

  &__header {
    margin-bottom: $grid-gutter-width;
    text-align: center;

    .app__logo {
      svg {
        width: 25rem;
      }
    }
  }

  &__libraries .app__container {
    background-color: $white;
    padding-bottom: $grid-gutter-width;
    padding-top: $grid-gutter-width * 2;

    @include lg {
      padding-bottom: $grid-gutter-width * 2;
      padding-top: $grid-gutter-width * 2;
    }

    h2 {
      margin-bottom: $grid-gutter-width * 2;
      text-align: center;
    }

    ul {
      display: flex;
      flex-wrap: wrap;
      list-style: none;
      padding: 0;

      > li {
        width: percentage(1 / 2);

        @include md {
          width: percentage(1 / 3);
        }
      }
    }
  }

  &__library {
    display: flex;
    flex-direction: column;
    margin-bottom: $grid-gutter-width;
    text-align: center;

    &__image {
      align-items: center;
      display: flex;
      justify-content: center;
      margin-bottom: $grid-gutter-width / 3;
      min-height: 10rem;

      svg {
        max-height: 10rem;
        width: 50%;
      }
    }
  }
}
