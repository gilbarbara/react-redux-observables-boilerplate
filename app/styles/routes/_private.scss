.app__private {
  h2 {
    margin-bottom: $grid-gutter-width;
  }

  &__running {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    min-height: 20vh;
  }

  &__repos {
    > div {
      a {
        font-size: $font-size-h5;
        margin-bottom: $grid-gutter-width / 6;
      }

      + div {
        margin-top: $grid-gutter-width / 3;
      }
    }
  }
}
