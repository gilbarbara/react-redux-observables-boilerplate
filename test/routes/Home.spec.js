import React from 'react';
import { mount } from 'enzyme';

import { Home } from 'routes/Home';

const mockDispatch = jest.fn();

function setup() {
  const props = {
    dispatch: mockDispatch,
    location: {},
  };

  return mount(<Home {...props} />);
}

describe('Home', () => {
  const wrapper = setup(true);

  it('should be a Component', () => {
    expect(wrapper.instance() instanceof React.Component).toBe(true);
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
