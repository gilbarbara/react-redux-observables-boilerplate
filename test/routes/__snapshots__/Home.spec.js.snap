// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render properly 1`] = `"<div class=\\"app__home app__route\\"><div class=\\"app__home__intro\\"><div class=\\"app__container\\"><div class=\\"app__home__header\\"><div class=\\"app__logo\\"><span class=\\"isvg failed\\"><img src=\\"\\" alt=\\"React-Redux-Observables Boilerplate\\"></span></div></div><h1>React-Redux-Observables Boilerplate</h1><p>A starter kit for React with Router, Redux, Observable + RxJS</p><a href=\\"https://github.com/gilbarbara/react-redux-observables-boilerplate\\" class=\\"app__home__download btn btn-lg btn-primary btn-icon\\" target=\\"_blank\\"><i class=\\"i-github\\"></i><span>Get it!</span></a></div></div><div class=\\"app__home__libraries\\"><div class=\\"app__container\\"><h2>Provides</h2><ul><li><div class=\\"app__home__library\\"><div class=\\"app__home__library__image\\"><span class=\\"isvg failed\\"></span></div><div class=\\"app__home__library__content\\"><h4>React</h4></div></div></li><li><div class=\\"app__home__library\\"><div class=\\"app__home__library__image\\"><span class=\\"isvg failed\\"></span></div><div class=\\"app__home__library__content\\"><h4>Redux</h4></div></div></li><li><div class=\\"app__home__library\\"><div class=\\"app__home__library__image\\"><span class=\\"isvg failed\\"></span></div><div class=\\"app__home__library__content\\"><h4>Redux Observable</h4></div></div></li><li><div class=\\"app__home__library\\"><div class=\\"app__home__library__image\\"><span class=\\"isvg failed\\"></span></div><div class=\\"app__home__library__content\\"><h4>react-router</h4></div></div></li><li><div class=\\"app__home__library\\"><div class=\\"app__home__library__image\\"><span class=\\"isvg failed\\"></span></div><div class=\\"app__home__library__content\\"><h4>RxJS</h4></div></div></li><li><div class=\\"app__home__library\\"><div class=\\"app__home__library__image\\"><span class=\\"isvg failed\\"></span></div><div class=\\"app__home__library__content\\"><h4>Webpack 2.x</h4></div></div></li></ul></div></div></div>"`;
